src/objects/zcl_abapgit_object_bdef.clas.abap

```diff
@@ -204,72 +274,101 @@ CLASS zcl_abapgit_object_bdef IMPLEMENTATION.
-          fill_metadata_from_db( CHANGING cs_behaviour_definition = <ls_behaviour_definition> ).
-          li_object_data_model->set_data( <ls_behaviour_definition> ).
+        CREATE DATA lr_wbobjtype TYPE ('WBOBJTYPE').
+        ASSIGN lr_wbobjtype->* TO <ls_wbobjtype>.
+        ASSIGN COMPONENT 'OBJTYPE_TR' OF STRUCTURE <ls_wbobjtype> TO <lv_field>.
+        <lv_field> = 'BDEF'.
+        ASSIGN COMPONENT 'SUBTYPE_WB' OF STRUCTURE <ls_wbobjtype> TO <lv_field>.
+        <lv_field> = 'BDO'.
 
-          CALL METHOD li_wb_object_operator->('IF_WB_OBJECT_OPERATOR~UPDATE')
-            EXPORTING
-              io_object_data    = li_object_data_model
-              transport_request = lv_transport_request.
+        CREATE DATA lr_category TYPE ('WBADT_RESOURCE_CATEGORY').
+        ASSIGN lr_category->* TO <lv_category>.
 
-        ELSE.
+        CALL METHOD ('CL_BLUE_WB_UTILITY')=>('GET_RESOURCE_CATEGORY')
+          EXPORTING
+            is_object_type = <ls_wbobjtype>
+          RECEIVING
+            result         = <lv_category>.
 
-          li_object_data_model->set_data( <ls_behaviour_definition> ).
+        lo_wb_object_operator = get_wb_object_operator( ).
 
-          CALL METHOD li_wb_object_operator->('IF_WB_OBJECT_OPERATOR~CREATE')
-            EXPORTING
-              io_object_data    = li_object_data_model
-              data_selection    = 'P' " if_wb_object_data_selection_co=>c_properties
-              package           = iv_package
-              transport_request = lv_transport_request.
+        tadir_insert( iv_package ).
 
-          CALL METHOD li_wb_object_operator->('IF_WB_OBJECT_OPERATOR~UPDATE')
-            EXPORTING
-              io_object_data    = li_object_data_model
-              data_selection    = 'D' " if_wb_object_data_selection_co=>c_data_content
-              transport_request = lv_transport_request.
 
+        IF zif_abapgit_object~exists( ) = abap_false.
+          CASE <lv_category>.
+            WHEN '1'. "if_wb_adt_plugin_resource_co=>co_sfs_res_category_atomic.
+              CALL METHOD lo_wb_object_operator->('IF_WB_OBJECT_OPERATOR~CREATE')
+                EXPORTING
+                  io_object_data    = lo_object_data
+                  data_selection    = 'AL' " if_wb_object_data_selection_co=>c_all_data
+                  version           = 'I'
+                  package           = iv_package
+                  transport_request = lv_transport_request.
+            WHEN '2'. "if_wb_adt_plugin_resource_co=>co_sfs_res_category_compound_s.
+              CALL METHOD lo_wb_object_operator->('IF_WB_OBJECT_OPERATOR~CREATE')
+                EXPORTING
+                  io_object_data    = lo_object_data
+                  data_selection    = 'P' " if_wb_object_data_selection_co=>c_properties
+                  version           = 'I'
+                  package           = iv_package
+                  transport_request = lv_transport_request.
```

Comment:
@ThomasPloski Next error
![image](https://user-images.githubusercontent.com/17437789/96251135-50408b00-0fb0-11eb-85f8-836546ed15db.png)

![image](https://user-images.githubusercontent.com/17437789/96251143-52a2e500-0fb0-11eb-8bc9-fcf11762e4ae.png)

Reply:
@ThomasPloski

Reply:
@ThomasPloski

Reply:
I guess this was fixed in the commits below? @christianguenter2 can you try again?